### PR TITLE
ref(api): Move ProjectGroupIndexEndpoint.put into helper to support multi project

### PR DESCRIPTION
**Pull request by macqueen**
~~Going to spot check / run through this again on Tuesday to make sure I haven't missed anything, but figured i'd put this up for initial feedback.~~

ok this should be good for review now

### CODE REVIEW

File: src/sentry/api/helpers/group_index.py [hunk 1]
```diff
@@ -319,3 +334,558 @@ def delete_groups(request, project, group_list, delete_type):
...
+            reason=GroupSubscriptionReason.status_change,
+        )
+        self_assign_issue = UserOption.objects.get_value(
+            user=acting_user, key='self_assign_issue', default='0'
+        )
+        if self_assign_issue == '1' and not group.assignee_set.exists():
+            return Actor(type=User, id=acting_user.id)
+
+
+def update_groups(request, projects, organization_id, search_fn):
```
Comment (macqueen) [inline src/sentry/api/helpers/group_index.py]: projects here would be the projects a user has access to within the org. I will probably add something within this function that filters that down to the projects that actually belong to the groups we're updating as I think that will help the efficiency in some cases. Will do that before adding to the org endpoint but in a separate diff than this.

File: src/sentry/api/helpers/group_index.py [hunk 2]
```diff
@@ -319,3 +334,558 @@ def delete_groups(request, project, group_list, delete_type):
...
+            status_details = {
+                'inRelease': release.version,
+                'actor': serialize(extract_lazy_object(request.user), request.user),
+            }
+            res_type = GroupResolution.Type.in_release
+            res_type_str = 'in_release'
+            res_status = GroupResolution.Status.resolved
+        elif statusDetails.get('inCommit'):
+            # TODO(jess): Same here, this is probably something we could do, but
+            # punting for now.
```
Comment (markstory) [inline src/sentry/api/helpers/group_index.py]: The UI also attempts to prevent this state from being created.

**Review by wedamija — APPROVED** [review]
Dang this code is complicated to follow, looks good since this is mostly just moving to a helper

File: src/sentry/api/helpers/group_index.py [hunk 3]
```diff
@@ -319,3 +334,558 @@ def delete_groups(request, project, group_list, delete_type):
...
+        # if we've specified a commit, let's see if its already been released
+        # this will allow us to associate the resolution to a release as if we
+        # were simply using 'inRelease' above
+        # Note: this is different than the way commit resolution works on deploy
+        # creation, as a given deploy is connected to an explicit release, and
+        # in this case we're simply choosing the most recent release which contains
+        # the commit.
+        if commit and not release:
+            try:
+                release = Release.objects.filter(
```
[the remark this answers is not in the document]
Comment (macqueen) [reply src/sentry/api/helpers/group_index.py]: yeah i thought about this too, but good call leaving a comment, will do

[the remark this answers is not in the document]
Comment (macqueen) [reply src/sentry/api/helpers/group_index.py]: added above

**Pull request merged**